core/services/pipeline/orm.go

```diff
@@ -136,7 +136,7 @@ func (o *orm) StoreRun(tx postgres.Queryer, run *Run) (restart bool, err error)
 
 		// Suspend the run
 		run.State = RunStatusSuspended
-		if _, err = sqlx.NamedExec(tx, `UPDATE pipeline_runs SET state = :state`, run); err != nil {
+		if _, err = sqlx.NamedExec(tx, `UPDATE pipeline_runs SET state = :state WHERE id = :id`, run); err != nil {
```

Comment:
Took me a while to find this one - seems quite important too cc @samsondav @archseer. Popped up because my integration test does a second ethtx task job after one is already complete, without this "where id" it tries to mark all the pipeline runs suspended and hits the pipeline_runs_check constraint on the already completed one.

Reply:
Good catch.. I can't believe I missed it

Reply:
Uff that's a nasty one